### PR TITLE
CORTX-32355 Server Pod takes more than 10 minutes to start when started concurrently with data pod

### PR DESCRIPTION
**Pull request by madhavemuri**
Problem: rgw admin util fails with timeout when started before motr ios mkfs is done

    When both data and server pods are started at once, mkfs of ioservice processes
    done in init contianer of data pods is not yet complete, but in init container of
    RGW server pods rgw admin util try to create index and user account in it.
    This causes failures of the utility as mkfs and m0d shares the same endpoint.

    Solution:
    Make sure that mkfs done for all isoervice is checked in init phase of hare
    in RGW server init containers as well. This is a one time activity per deployment.

### CODE REVIEW

**Comment by vaibhavparatwar** [comment]
@madhavemuri  Request you to link the PR with JIRA and add appropriate commit message. Also, please update your branch with latest main.

**Comment by madhavemuri** [comment]
```[root@ssc-vm-g4-rhev4-1532 k8_cortx_cloud]# kubectl get pods -n reallylongnamespace
NAME                             READY   STATUS    RESTARTS       AGE
cortx-consul-client-dfxg4        1/1     Running   0              2m14s
cortx-consul-server-0            1/1     Running   0              2m14s
cortx-control-76d64b9ccc-lpg9n   1/1     Running   0              2m14s
cortx-data-g0-0                  3/3     Running   0              2m13s
cortx-data-g1-0                  3/3     Running   0              2m14s
cortx-ha-ccb69c8b7-6vx6k         3/3     Running   0              2m14s
cortx-kafka-0                    1/1     Running   1 (115s ago)   2m14s
cortx-server-0                   2/2     Running   0              2m14s
cortx-server-1                   2/2     Running   0              2m14s
cortx-zookeeper-0                1/1     Running   0              2m14s
[root@ssc-vm-g4-rhev4-1532 k8_cortx_cloud]#```

Tested locally deployment completed within 2 mins.

**Comment by madhavemuri** [comment]
> @madhavemuri Request you to link the PR with JIRA and add appropriate commit message. Also, please update your branch with latest main.

Done.

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -551,13 +548,15 @@ def init(args):
...
                                        config_dir, log_dir, url)
-        hax_starter = _start_hax(utils, stop_event, config_dir, log_dir)
         hostname = utils.get_local_hostname()
-        # Cleanup old mkfs state
-        cleanup_mkfs_state(utils, cns_utils)
-        start_mkfs_parallel(hostname, config_dir)
-        # Update mkfs state
-        set_mkfs_done_for(hostname, cns_utils)
+        if is_mkfs_required(url):
+            hax_starter = _start_hax(utils, stop_event, config_dir, log_dir)
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/main.py:553]: we still need to start hax in server pod, why is it avoided?

**Comment by SwapnilGaonkar7** [comment]
@madhavemuri I think Jira number is still not there in commit message.
Can you please check.

File: provisioning/miniprov/hare_mp/main.py (hunk 1 above)
Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/main.py:553]: @mssawant In 'init' stage?

**Review by mssawant — APPROVED** [review]
Looks good.

**Comment by mssawant** [comment]
@SwapnilGaonkar7, please ignore. I deleted the comment.

**Comment by SwapnilGaonkar7** [comment]
retest this please

**Comment by vaibhavparatwar** [comment]
@madhavemuri  1N / 3N deployment seems to be failing for this. cc @mssawant

**Review by Shreya-18 — APPROVED** [review]

**Comment by SwapnilGaonkar7** [comment]
There are 2 commits, there needs to be only 1

**Review by d-nayak — APPROVED** [review]

**Pull request merged**